### PR TITLE
client/web: add web client Server struct

### PR DESCRIPTION
**Pull request by soniaappasamy**
Updates tailscale/corp#13775

### CODE REVIEW

File: client/web/web.go [hunk 1]
```diff
@@ -46,6 +46,20 @@ var tmpl *template.Template
 
 var localClient tailscale.LocalClient
```
Comment (willnorris) [inline client/web/web.go:47]: I think we can remove this localClient now?

**Review by willnorris — APPROVED** [review]

File: client/web/web.go [hunk 2]
```diff
@@ -44,7 +44,19 @@ var authenticationRedirectHTML string
 
 var tmpl *template.Template
 
-var localClient tailscale.LocalClient
+// Server is the backend server for a Tailscale web client.
+type Server struct {
+	devMode bool
```
Comment (willnorris) [inline client/web/web.go:49]: What about exporting these fields, and just letting callers set them directly?  For example, in the tsnet use case, the caller is going to need to set the `*tailscale.LocalClient` to the one associated with the tsnet server.  So something like:

``` go
s := new(tsnet.Server)
lc, _ := s.LocalClient()
ws := &web.Server{
    LocalClient: lc,
}
```

File: client/web/web.go [hunk 3]
```diff
@@ -265,7 +277,7 @@ req.send(null);
 `
 
 // Handle processes all requests for the Tailscale web client.
-func Handle(w http.ResponseWriter, r *http.Request) {
+func (s *Server) Handle(w http.ResponseWriter, r *http.Request) {
```
Comment (willnorris) [inline client/web/web.go:280]: oh, if we're making this a method on the Server type, then we should rename it to `ServeHTTP` so then it implements the `http.Handler` interface directly.  So then callers could just do:

```go
ws := new(web.Server)
http.ListenAndServe(":8080", ws)
```

Comment (soniaappasamy) [reply client/web/web.go:280]: nice okay!

File: client/web/web.go (hunk 2 above)
Comment (soniaappasamy) [reply client/web/web.go:49]: thoughts on having `NewServer` accept a `*tailscale.LocalClient` instead?

Comment (willnorris) [reply client/web/web.go:49]: yeah, that could certainly work too.  I suspect we have prior art for both throughout the codebase.  I'm just not sure how long this list of method params would grow.  Maybe not too bad.

Comment (willnorris) [reply client/web/web.go:49]: I was trying to think about the the zero value `web.Server` being usable.  But if LocalClient is a pointer, then that doesn't really work.  So in that case, I guess a New method is clearer in how it should be used.

Comment (soniaappasamy) [reply client/web/web.go:49]: That way if we ever have special stuff in there that callers don't need to provide then we don't have to switch stuff over. Could even make the local client arg optional...

File: cmd/tailscale/cli/web.go [hunk 4]
```diff
@@ -76,7 +78,8 @@ func runWeb(ctx context.Context, args []string) error {
 		return fmt.Errorf("too many non-flag arguments: %q", args)
 	}
 
-	webHandler := http.HandlerFunc(web.Handle)
+	webServer := web.NewServer(webArgs.dev, nil)
+	webHandler := http.HandlerFunc(webServer.ServeHTTP)
```
Comment (willnorris) [inline cmd/tailscale/cli/web.go:82]: there's no need for `webHandler` at all now.  You can just pass webServer directly wherever you need an `http.Handler` below.  e.g.

```go
cgi.Serve(webServer)
```

**Review by willnorris — APPROVED** [review]
one final simplification, but LGTM

**Pull request merged**